app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -266,9 +268,30 @@ object TelemetryWrapper {
         TelemetryEvent.create(Category.ACTION, Method.CLICK, Object.MENU, button.value).queue()
     }
 
-    @JvmStatic
-    fun menuAppNavEvent(button: MenuAppNavButton) {
-        TelemetryEvent.create(Category.ACTION, Method.CLICK, Object.MENU, button.value).queue()
+    fun overlayClickEvent(event: NavigationEvent, isTurboButtonChecked: Boolean, isPinButtonChecked: Boolean) {
```

Comment:
Hmm, I wonder if we can bundle `is*Checked` into the `NavigationEvent`? Maybe more sealed classes, but maybe overkill.

Reply:
You can't assign values to enums (they're singletons) so my desired case is: `onNavigationEvent(event, /* any params necessary */)`. This doesn't work though: NavigationEvent is overloaded so that it gets called for any Overlay -> BrowserFragment callsbacks, so e.g. a `LOAD` or `RELOAD_YT` event needs to call onNavigationEvent but doesn't have the parameters to do so.

In theory, we could do `onNavigationEvent(event, someSealedClassArgs)` but that seems like an unnecessary abstraction.

Fun fact: Swift solves this sealed class/enum problem by letting you assign values to enums. What do you know? :)